.github/workflows/test.yml

```diff
@@ -232,3 +232,56 @@ jobs:
 
       - name: Run Forge example tests using remote network with a pinned block number
         run: forge test --remappings hedera-forking=. -vv --fork-url https://mainnet.hashio.io/api --fork-block-number 72433403
+
+  sol-tests-coverage:
+    permissions:
+      pull-requests: write
+    name: 'Solidity Coverage test'
+    runs-on: smart-contracts-linux-medium
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@5c7944e73c4c2a096b17a9cb74d65b6c2bbafbde # v2.9.1
+        with:
+          egress-policy: audit
+
+      - name: Checkout Code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          submodules: recursive
+
+      - name: Set up Node.js
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4.0.3
+        with:
+          node-version: '20'
+
+      - name: Install Foundry
+        uses: step-security/foundry-toolchain@01e2d127ea061e2ce8fa34340de71ca8518f093e # v1.2.1
+        with:
+          version: nightly
+
+      - name: Run Forge build
+        run: |
+          forge --version
+          forge build
+
+      - name: Generate Coverage Report
+        run: |
+          forge coverage --report lcov
+          cat lcov.info
+
+      - name: Convert LCOV to Istanbul Text Format
+        run: |
+          .github/lcovtoistanbul lcov.info > istanbul.txt
```

Comment:
instead of manually converting the report, shouldn't we use a package to do this? See dependencies of  https://github.com/bcoe/c8, we might find something useful. I mentioned `c8` because that's what we will usef or JS coverage.